group/edwards25519/point.go

```diff
@@ -290,3 +296,319 @@ func (P *point) IsCanonical(s []byte) bool {
 
 	return 1-(c&d&1) == 1
 }
+
+func (P *point) Hash(m []byte, dst string) kyber.Point {
+	u := hashToField(m, dst, 2)
+	q0 := mapToCurveElligator2Ed25519(u[0])
+	q1 := mapToCurveElligator2Ed25519(u[1])
+	P.Add(q0, q1)
+
+	// Clear cofactor
+	P.Mul(cofactorScalar, P)
+
+	return P
+}
+
+func hashToField(m []byte, dst string, count int) []fieldElement {
+	l := 48 // L param in RFC9380
+	byteLen := count * l
+	uniformBytes, _ := expandMessageXMD(sha512.New(), m, dst, byteLen)
+
+	u := make([]fieldElement, count)
+	for i := 0; i < count; i++ {
+		elmOffset := l * i
+		tv := big.NewInt(0).SetBytes(uniformBytes[elmOffset : elmOffset+l])
+		tv = tv.Mod(tv, prime)
+		fe := fieldElement{}
+		feFromBn(&fe, tv)
+		u[i] = fe
+	}
+
+	return u
+}
+
+func expandMessageXMD(h hash.Hash, m []byte, domainSeparator string, byteLen int) ([]byte, error) {
+	r := float64(byteLen) / float64(h.Size()>>3)
+	ell := int(math.Ceil(r))
+	if ell > 255 || ell < 0 || byteLen > 65535 {
```

Comment:
The RFC specifies that the domain separator MUST have nonzero length: I'd add this to the invalid parameters test (cf. Section 3.1 in the RFC).

Reply:
Done !